src/app/core/store/effects/auth.effects.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from "@angular/core";
+import { Actions, ofType, createEffect } from "@ngrx/effects";
+import { Router } from "@angular/router";
+
+import { catchError, exhaustMap, switchMap, take, tap } from "rxjs/operators";
+import { Observable, of } from "rxjs";
+
+import { IUserLogin, IUser, IAuthResponse } from "../../models/index";
+import { AuthService } from "../../services/index";
+import { StoreFacadeService } from "../../store-facade/store-facade.service";
+
+import * as AuthActions from "../actions/auth.actions";
+
+
+@Injectable()
+export class AuthEffects {
+
+
+    constructor(
+        private actions$: Actions,
+        private router: Router,
+        private authService: AuthService,
+        private storeFacadeService: StoreFacadeService
+    ) {
+    }
+
+
+    public authLogin$: Observable<any> = createEffect(

```

Comment:
declare properties before the constructor and assign in constructor

Reply:
done